### DISCUSSION TITLE
Log Backtrace / Stacktrace with a custom ErrorHandler

### DISCUSSION THREAD

**Discussion by vincent-herlemont** [fzyzcjy/flutter_rust_bridge#740] [Q&A]
Is possible to return the line number/backtrace or event custom backtrace when we log errors with `CustomErrorHandler`?

NOTE: Some libraries allow to customize stack trace and offer divers report improvements like [miette](https://docs.rs/miette/latest/miette/index.html#), [stable_eyre](https://docs.rs/stable-eyre/latest/stable_eyre/) etc.  There is some method to retrieve manually the line number when the program panic here: [Getting line numbers with `?` as I would with `unwrap()`](https://users.rust-lang.org/t/getting-line-numbers-with-as-i-would-with-unwrap/47002).

#### More context

Below is an example to log a panic produces by `unreachable!()`.
```rust
#[derive(Clone, Copy)]
pub struct CustomErrorHandler(pub handler::ReportDartErrorHandler);

impl handler::ErrorHandler for CustomErrorHandler {
    fn handle_error(&self, port: i64, error: handler::Error) {
        logs::info(&error.message());
        self.0.handle_error(port, error)
    }
...

pub fn failed_function() {
    unreachable!("throw error")
}
```

We can see it that only the message of `throw error ` is displayed.
```bash
I/flutter ( 6694): log from rust: internal error: entered unreachable code: throw error 
```
In this case, it should be greater to retrieve the file name and line number. Here is the goal of this discussion.

---
Related to:
* [Support for Result<MyType, MyError> with exception throwing on Dart by lattice0 · Pull Request #582](https://github.com/fzyzcjy/flutter_rust_bridge/pull/582)
* [\`Result<(),MyCustomError>\` instead of \`anyhow\`? · Issue #533](https://github.com/fzyzcjy/flutter_rust_bridge/issues/533)

**Comment by fzyzcjy** [comment]
I personally use sentry and it knows the stack trace IIRC.

**Reply by vincent-herlemont** [reply]
@fzyzcjy Thanks for your answer. I will try it. However, in a full offline configuration, it should be greater to have readable error logs with the most details as possible.

Is it possible to provide a custom [`handler::Error`](https://github.com/fzyzcjy/flutter_rust_bridge/blob/7691954f0d1b0c0aa7d414ac2fbe9f866c8c57fc/frb_rust/src/handler.rs#L259-L290) in the same way that we can purpose a custom implementation of `handler::ErrorHandler`?

**Reply by fzyzcjy** [reply]
Just look at sentry's source code, its client is OSS
